src/util/records.py

```diff
@@ -272,24 +273,28 @@ def get_trans_terms(value):
             )
 
 
-def add_view(cr, name, model, view_type, arch_db, inherit_xml_id=None, priority=16):
+def add_view(cr, name, model, view_type, arch_db, inherit_xml_id=None, priority=16, key=""):
     inherit_id = None
     if inherit_xml_id:
         inherit_id = ref(cr, inherit_xml_id)
         if not inherit_id:
             raise ValueError(
                 "Unable to add view '%s' because its inherited view '%s' cannot be found!" % (name, inherit_xml_id)
             )
+    # Odoo <= 8.0 doesn't have the `key`
+    key_exist = column_exists(cr, "ir_ui_view", "key")
+    if key_exist and view_type == "qweb" and not key:
+        key = "gen_key.%s" % str(uuid.uuid4())[:6]
     arch_col = "arch_db" if column_exists(cr, "ir_ui_view", "arch_db") else "arch"
     jsonb_column = column_type(cr, "ir_ui_view", arch_col) == "jsonb"
     arch_column_value = Json({"en_US": arch_db}) if jsonb_column else arch_db
     cr.execute(
         """
         INSERT INTO ir_ui_view(name, "type",  model, inherit_id, mode, active, priority, %s)
-        VALUES(%%(name)s, %%(view_type)s, %%(model)s, %%(inherit_id)s, %%(mode)s, 't', %%(priority)s, %%(arch_db)s)
+        VALUES(%%(name)s, %%(view_type)s, %%(model)s, %%(inherit_id)s, %%(mode)s, 't', %%(priority)s, %%(arch_db)s %s)
         RETURNING id
     """
-        % arch_col,
+        % (arch_col + (", key" if key_exist else ""), ", '%s'" % key if key_exist else ""),
```

Comment:
This is not readable :/

At least, don't inject an unsafe variable in the query.
```suggestion
        % (arch_col + (", key" if key_exist else ""), ", %(key)s" if key_exist else ""),
```

Reply:
updated, Thank you for your input :slightly_smiling_face: